wasm/src/lib.rs

```diff
@@ -305,7 +305,7 @@ unsafe fn _decode_from_raw_in_range<T: Decode>(
     let bytes = Box::from_raw(core::slice::from_raw_parts_mut(ptr as *mut _, len));
```

Comment:
there are a few places in this module that you should replace `Decode` bound with `DecodeAll` as you did in `smartcontracts/wasm.rs`

Reply:
also check if `DumpDecoded` in `tools/parity_scale_codec` should be modified